Lib/email/feedparser.py

```diff
@@ -534,3 +559,85 @@ class BytesFeedParser(FeedParser):
 
     def feed(self, data):
         super().feed(data.decode('ascii', 'surrogateescape'))
+
+
+class NonIdentityTransferDecodingFeedParser(abc.ABC):
+    """
```

Comment:
hanging indent on the docstring looks strange to me; consult [PEP 257](https://www.python.org/dev/peps/pep-0257/)